Firestore/core/src/remote/bloom_filter.cc

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// TODO(Mila): Replace this CommonCrypto with platform based MD5 calculation.
+#pragma clang diagnostic push
```

Comment:
Use `SUPPRESS_DEPRECATED_DECLARATIONS_BEGIN()` and `SUPPRESS_END()` rather than using `#pragma` directly. This is more portable, and will also work in gcc.

e.g. https://github.com/firebase/firebase-ios-sdk/blob/ff50941bf77cb02979b29908c8ee3baa7dda4358/Firestore/core/src/util/md5_commoncrypto.cc#L37